angularjs/angular-animate.d.ts

```diff
@@ -14,27 +14,68 @@ declare module "angular-animate" {
  * ngAnimate module (angular-animate.js)
  */
 declare module angular.animate {
-    interface IAnimateFactory extends Function {
-        enter?: (element: ng.IAugmentedJQuery, doneFn: Function) => IAnimateCssRunner|void;
-        leave?: (element: ng.IAugmentedJQuery, doneFn: Function) => IAnimateCssRunner|void;
-        addClass?: (element: ng.IAugmentedJQuery, className: string, doneFn: Function) => IAnimateCssRunner|void;
-        removeClass?: (element: ng.IAugmentedJQuery, className: string, doneFn: Function) => IAnimateCssRunner|void;
-        setClass?: (element: ng.IAugmentedJQuery, className: string, doneFn: Function) => IAnimateCssRunner|void;
+    interface IAnimateFactory {
+        (...args: any[]): IAnimateCallbackObject;
     }
 
+    interface IAnimateCallbackObject {
+        eventFn?: (element: IAugmentedJQuery, doneFunction: Function, options: IAnimationOptions) => any;
+        setClass?: (element: IAugmentedJQuery, addedClasses: string, removedClasses: string, doneFunction: Function, options: IAnimationOptions) => any;
+        addClass?: (element: IAugmentedJQuery, addedClasses: string, doneFunction: Function, options: IAnimationOptions) => any;
+        removeClass?: (element: IAugmentedJQuery, removedClasses: string, doneFunction: Function, options: IAnimationOptions) => any;
+        enter?: (element: IAugmentedJQuery, doneFunction: Function, options: IAnimationOptions) => any;
+        leave?: (element: IAugmentedJQuery, doneFunction: Function, options: IAnimationOptions) => any;
+        move?: (element: IAugmentedJQuery, doneFunction: Function, options: IAnimationOptions) => any;
+        animate?: (element: IAugmentedJQuery, fromStyles: string, toStyles: string, doneFunction: Function, options: IAnimationOptions) => any;
```

Comment:
well, the `return value;` statement is inside an `isFunction` conditional... so I think you were right originally in that [`executeAnimationFn`](https://github.com/angular/angular.js/blob/master/src/ngAnimate/animateJs.js#L144-L185) always returns a function. I assume that `noop` is defined somewhere else as a function that does nothing.

also I'm not sure it ever returns a `$$AnimateRunner`... that is handled on L177 and it just sets `value.done` instead of returning